### ISSUE TITLE
Zathura launching fails

### ISSUE THREAD

**Issue by sebastinas** [pwmt/zathura#94]
*On GitLab by @goduck777 on Nov 1, 2018, 21:38*

---

Recently after upgrading to the latest Zathura (0.4.1), I cannot launch it anymore. The error I got is

error: Could not initialize zathura.

I also tried to start it with --log-level-debug, the no more helpful information. The system is Arch Linux.

**Comment by sebastinas** [comment]
*On GitLab by @goduck777 on Nov 1, 2018, 21:44*

---

I tried to downgrade to 0.4.0 and it can launch without issues.

**Comment by sebastinas** [comment]
Please post the output when running with `-l debug`. It should give us a rough idea where initialization fails. Or upgrade to `zathura-git` which should print a message indicating why it fails.

**Comment by sebastinas** [comment]
*On GitLab by @goduck777 on Nov 2, 2018, 15:52*

---

```
$zathura --log-level=debug
debug: ../girara/template.c:310: base_changed(): Variable 'session' not set.
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/pdf
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/oxps
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/epub+zip
debug: ../zathura/plugin.c:170: load_plugin(): Successfully loaded plugin from '/usr/lib/zathura/libpdf-mupdf.so'.
debug: ../zathura/plugin.c:173: load_plugin(): plugin pdf-mupdf: version 0.3.4
debug: ../zathura/zathura.c:435: zathura_init(): Basic sandbox allowing normal operation.
error: Could not initialize zathura.
```

I will try the git version later.

**Comment by sebastinas** [comment]
*On GitLab by @goduck777 on Nov 2, 2018, 23:08*

---

I see. The sandbox feature is not working. Adding "set sandbox none" in zathurarc fixed the issue.

**Comment by sebastinas** [comment]
Okay, thank you. @valoq, could to take a look at this issue?

**Comment by sebastinas** [comment]
*On GitLab by @valoq on Dec 19, 2018, 20:27*

---

Thank you for reporting this issue @goduck777 

Can you please give us some more information about your system please.

What is the output of the following commands: 

ldd /usr/bin/zathura | grep seccomp


Are you running a self compiled kernel by any chance?


So far I could not replicate this issue on arch linux.

The only reason for this issue I can think of is missing seccomp support in the kernel (very unlikely) or  missing libseccomp as a build dependency, which does not seem to be the case either as meson will take care of it and the zathura package on arch (build date 2018-10-20) works fine on my test system and links to libseccomp.

Also I do not recall any recent changes that could explain why the git version behaves differently regarding seccomp.

**Comment by sebastinas** [comment]
*On GitLab by @ertwro on Jan 17, 2019, 03:56*

---

Hi, I have a similar issue. In my case I use the wsl version of arch linux. When I use the current git version, or any version above 0.4.0-4 I get:

```
$ ldd /usr/bin/zathura | grep seccomp
libseccomp.so.2 => /usr/lib/libseccomp.so.2 (0x00007ffaab0e0000)
```

otherwise, the command returns as empty. That file is a symlink to ```/usr/lib/libseccomp.so.2.3.3```

**Comment by sebastinas** [comment]
*On GitLab by @valoq on Jan 20, 2019, 11:24*

---

Please post the output when running with `-l debug`

Version 0.4.0 and below do not link to libseccomp.

@ertwro 
Which version does not launch in your case? Does it work if you `set sandbox none` ?

Regarding wsl I think there might be a lot of potential causes for this issue. 
There are quite some reports about issues with applications using seccomp when running in wsl, therefore this seems like a Microsoft issue. Not sure zathura should deal with this.

**Comment by sebastinas** [comment]
*On GitLab by @ertwro on Jan 23, 2019, 01:35*

---

Anything above 0.4.0 doesn't work for me, while 0.4.0 does. Yes, adding `set sandbox none` into my zathurarc file does work. At first, I thought it was a problem exclusive to the kernel but after digging a little bit it almost looks like at some point form 0.4.1 on some dependencies in Zathura changed.

While this probably goes beyond the Zathura developer's scope in order to solve, it seems like it was due to a change in Zathura's dependencies and turning off the sandbox works fine, so I don't see a big problem here. While not super secure it works.

This is my debbug output in case you still wanna look at it.
```
debug: ../girara/template.c:311: base_changed(): Variable 'session' not set.
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: image/vnd.djvu
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: image/vnd.djvu+multipage
debug: ../zathura/plugin.c:170: load_plugin(): Successfully loaded plugin from '/usr/lib/zathura/libdjvu.so'.
debug: ../zathura/plugin.c:173: load_plugin(): plugin djvu: version 0.2.8
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/pdf
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/oxps
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/epub+zip
debug: ../zathura/plugin.c:170: load_plugin(): Successfully loaded plugin from '/usr/lib/zathura/libpdf-mupdf.so'.
debug: ../zathura/plugin.c:173: load_plugin(): plugin pdf-mupdf: version 0.3.4
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/postscript
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/eps
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: application/x-eps
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: image/eps
debug: ../zathura/plugin.c:275: register_plugin(): plugin: filetype mapping added: image/x-eps
debug: ../zathura/plugin.c:170: load_plugin(): Successfully loaded plugin from '/usr/lib/zathura/libps.so'.
debug: ../zathura/plugin.c:173: load_plugin(): plugin ps: version 0.2.6
debug: ../zathura/zathura.c:440: zathura_init(): Basic sandbox allowing normal operation.
debug: ../zathura/seccomp-filters.c:51: seccomp_enable_basic_filter(): adding rule kill to _sysctl
debug: ../zathura/seccomp-filters.c:52: seccomp_enable_basic_filter(): adding rule kill to acct
debug: ../zathura/seccomp-filters.c:53: seccomp_enable_basic_filter(): adding rule kill to add_key
debug: ../zathura/seccomp-filters.c:54: seccomp_enable_basic_filter(): adding rule kill to adjtimex
debug: ../zathura/seccomp-filters.c:55: seccomp_enable_basic_filter(): adding rule kill to chroot
debug: ../zathura/seccomp-filters.c:56: seccomp_enable_basic_filter(): adding rule kill to clock_adjtime
debug: ../zathura/seccomp-filters.c:57: seccomp_enable_basic_filter(): adding rule kill to create_module
debug: ../zathura/seccomp-filters.c:58: seccomp_enable_basic_filter(): adding rule kill to delete_module
debug: ../zathura/seccomp-filters.c:59: seccomp_enable_basic_filter(): adding rule kill to fanotify_init
debug: ../zathura/seccomp-filters.c:60: seccomp_enable_basic_filter(): adding rule kill to finit_module
debug: ../zathura/seccomp-filters.c:61: seccomp_enable_basic_filter(): adding rule kill to get_kernel_syms
debug: ../zathura/seccomp-filters.c:62: seccomp_enable_basic_filter(): adding rule kill to get_mempolicy
debug: ../zathura/seccomp-filters.c:63: seccomp_enable_basic_filter(): adding rule kill to init_module
debug: ../zathura/seccomp-filters.c:64: seccomp_enable_basic_filter(): adding rule kill to io_cancel
debug: ../zathura/seccomp-filters.c:65: seccomp_enable_basic_filter(): adding rule kill to io_destroy
debug: ../zathura/seccomp-filters.c:66: seccomp_enable_basic_filter(): adding rule kill to io_getevents
debug: ../zathura/seccomp-filters.c:67: seccomp_enable_basic_filter(): adding rule kill to io_setup
debug: ../zathura/seccomp-filters.c:68: seccomp_enable_basic_filter(): adding rule kill to io_submit
debug: ../zathura/seccomp-filters.c:69: seccomp_enable_basic_filter(): adding rule kill to ioperm
debug: ../zathura/seccomp-filters.c:70: seccomp_enable_basic_filter(): adding rule kill to iopl
debug: ../zathura/seccomp-filters.c:71: seccomp_enable_basic_filter(): adding rule kill to ioprio_set
debug: ../zathura/seccomp-filters.c:72: seccomp_enable_basic_filter(): adding rule kill to kcmp
debug: ../zathura/seccomp-filters.c:73: seccomp_enable_basic_filter(): adding rule kill to kexec_file_load
debug: ../zathura/seccomp-filters.c:74: seccomp_enable_basic_filter(): adding rule kill to kexec_load
debug: ../zathura/seccomp-filters.c:75: seccomp_enable_basic_filter(): adding rule kill to keyctl
debug: ../zathura/seccomp-filters.c:76: seccomp_enable_basic_filter(): adding rule kill to lookup_dcookie
debug: ../zathura/seccomp-filters.c:77: seccomp_enable_basic_filter(): adding rule kill to mbind
debug: ../zathura/seccomp-filters.c:78: seccomp_enable_basic_filter(): adding rule kill to nfsservctl
debug: ../zathura/seccomp-filters.c:79: seccomp_enable_basic_filter(): adding rule kill to migrate_pages
debug: ../zathura/seccomp-filters.c:80: seccomp_enable_basic_filter(): adding rule kill to modify_ldt
debug: ../zathura/seccomp-filters.c:81: seccomp_enable_basic_filter(): adding rule kill to mount
debug: ../zathura/seccomp-filters.c:82: seccomp_enable_basic_filter(): adding rule kill to move_pages
debug: ../zathura/seccomp-filters.c:83: seccomp_enable_basic_filter(): adding rule kill to name_to_handle_at
debug: ../zathura/seccomp-filters.c:84: seccomp_enable_basic_filter(): adding rule kill to open_by_handle_at
debug: ../zathura/seccomp-filters.c:85: seccomp_enable_basic_filter(): adding rule kill to perf_event_open
debug: ../zathura/seccomp-filters.c:86: seccomp_enable_basic_filter(): adding rule kill to pivot_root
debug: ../zathura/seccomp-filters.c:87: seccomp_enable_basic_filter(): adding rule kill to process_vm_readv
debug: ../zathura/seccomp-filters.c:88: seccomp_enable_basic_filter(): adding rule kill to process_vm_writev
debug: ../zathura/seccomp-filters.c:89: seccomp_enable_basic_filter(): adding rule kill to ptrace
debug: ../zathura/seccomp-filters.c:90: seccomp_enable_basic_filter(): adding rule kill to reboot
debug: ../zathura/seccomp-filters.c:91: seccomp_enable_basic_filter(): adding rule kill to remap_file_pages
debug: ../zathura/seccomp-filters.c:92: seccomp_enable_basic_filter(): adding rule kill to request_key
debug: ../zathura/seccomp-filters.c:93: seccomp_enable_basic_filter(): adding rule kill to set_mempolicy
debug: ../zathura/seccomp-filters.c:94: seccomp_enable_basic_filter(): adding rule kill to swapoff
debug: ../zathura/seccomp-filters.c:95: seccomp_enable_basic_filter(): adding rule kill to swapon
debug: ../zathura/seccomp-filters.c:96: seccomp_enable_basic_filter(): adding rule kill to sysfs
debug: ../zathura/seccomp-filters.c:97: seccomp_enable_basic_filter(): adding rule kill to syslog
debug: ../zathura/seccomp-filters.c:98: seccomp_enable_basic_filter(): adding rule kill to tuxcall
debug: ../zathura/seccomp-filters.c:99: seccomp_enable_basic_filter(): adding rule kill to umount2
debug: ../zathura/seccomp-filters.c:100: seccomp_enable_basic_filter(): adding rule kill to uselib
debug: ../zathura/seccomp-filters.c:101: seccomp_enable_basic_filter(): adding rule kill to vmsplice
error: Failed to initialize basic seccomp filter.
error: Could not initialize zathura.
```

**Comment by sebastinas** [comment]
We could also detect if zathura is running under wsl. According to https://stackoverflow.com/questions/38859145/detect-ubuntu-on-windows-vs-native-ubuntu-from-bash-script, `/proc/version` contains `Microsoft` in this case. So if wsl is detected, we could set the default to `none` and document the issue in the manpage.

**Comment by sebastinas** [comment]
*On GitLab by @goduck777 on Feb 11, 2019, 23:17*

---

@sebastinas @valoq I am running zathura in wsl also, so I got the same issue with @ertwro. The Linux distro is Arch.
For now I think it is best to detect wsl and disable sandbox accordingly.

**Comment by sebastinas** [comment]
mentioned in commit 8f3d20de88b0582676172cea4a2053a1b66833dd

**Comment by sebastinas** [comment]
Fixed in 8f3d20de.

**Comment by sebastinas** [comment]
closed
